pkg/gardenlet/controller/seed/seed_care_control.go

```diff
@@ -0,0 +1,233 @@
+		return nil
+	}
+
+	staleExtensionHealthCheckThreshold := confighelper.StaleExtensionHealthChecksThreshold(r.config.Controllers.ShootCare.StaleExtensionHealthChecks)
+
+	//Trigger health check
+	shootHealth := NewHealthCheck(seedClient, log)
+	updatedConditions := shootHealth.CheckSeed(
+		ctx,
+		seedObj,
+		conditions,
+		r.conditionThresholdsToProgressingMapping(),
+		staleExtensionHealthCheckThreshold,
+	)
+
+	// Update Shoot status (conditions, constraints) if necessary
+	if gardencorev1beta1helper.ConditionsNeedUpdate(conditions, updatedConditions) {
+		// Rebuild shoot conditions and constraints to ensure that only the conditions and constraints with the
+		// correct types will be updated, and any other conditions will remain intact
+		conditions := buildSeedConditions(seed.Status.Conditions, updatedConditions, conditionTypes)
+		log.Debugf("[SEED CARE] Updating shoot status conditions and constraints")
+		if err := patchSeedStatus(ctx, gardenClient, seed, conditions); err != nil {
+			log.Errorf("Could not update Shoot status: %+v", err)
+			return nil // We do not want to run in the exponential backoff for the condition checks.
+		}
+	}
+	return nil
+}
+
+func careSetupFailure(ctx context.Context, gardenClient client.Client, seed *gardencorev1beta1.Seed, message string, conditions []gardencorev1beta1.Condition) error {
+
+	updatedConditions := make([]gardencorev1beta1.Condition, 0, len(conditions))
+	for _, cond := range conditions {
+		updatedConditions = append(updatedConditions, gardencorev1beta1helper.UpdatedConditionUnknownErrorMessage(cond, message))
+	}
+
+	if !gardencorev1beta1helper.ConditionsNeedUpdate(conditions, updatedConditions) {
+		return nil
+	}
+
+	return patchSeedStatus(ctx, gardenClient, seed, updatedConditions)
+}
+
+// buildSeedConditions builds and returns the shoot conditions using the given shoot conditions as a base,
+// by first removing all conditions with the given types and then merging the given conditions (which must be of the same types).
+func buildSeedConditions(seedConditions []gardencorev1beta1.Condition, conditions []gardencorev1beta1.Condition, conditionTypes []gardencorev1beta1.ConditionType) []gardencorev1beta1.Condition {
+	result := gardencorev1beta1helper.RemoveConditions(seedConditions, conditionTypes...)
+	result = gardencorev1beta1helper.MergeConditions(result, conditions...)
+	return result
+}
+
+func patchSeedStatus(ctx context.Context, c client.StatusClient, seed *gardencorev1beta1.Seed, conditions []gardencorev1beta1.Condition) error {
+	patch := client.StrategicMergeFrom(seed.DeepCopy())
+	seed.Status.Conditions = conditions
+	return c.Status().Patch(ctx, seed, patch)
+}
+
+func (r *careReconciler) conditionThresholdsToProgressingMapping() map[gardencorev1beta1.ConditionType]time.Duration {
+	out := make(map[gardencorev1beta1.ConditionType]time.Duration)
+	for _, threshold := range r.config.Controllers.ShootCare.ConditionThresholds {
```

Comment:
Either we introduce dedicated thresholds for `Seed`s or we reuse the `ShootCare` condition thresholds. In this case, however, I would vote for moving this out of `Controllers.ShootCare`. cc @timuthy

Reply:
This comment is still not resolved

Reply:
Even though the `SystemComponentsHealthy` condition is already present in the `shootCare` config I'd agree that it makes more sense to add an additional `seedCare` config.

Reply:
Done